component/Page/Profile/index.tsx

```diff
@@ -31,6 +31,7 @@ export default function ProfilePage(props: PageProps) {
     const router = useRouter();
     useEffect(() => {
         if (data) {
+            localStorage.setItem(MessageEnum.targetUserName, username);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Storing sensitive data like usernames in local storage can lead to potential security risks such as XSS (Cross-Site Scripting) attacks. Consider using HttpOnly cookies or secure server-side sessions for storing such information.
```diff
- localStorage.setItem(MessageEnum.targetUserName, username);
+ // Consider a more secure method to store the username
```
Also, it's important to handle the case where `localStorage` is not available, such as in incognito mode in some browsers. This could cause the application to crash. You should add error handling around the `localStorage.setItem` calls.
```diff
- localStorage.setItem(MessageEnum.targetUserName, username);
+ try {
+     localStorage.setItem(MessageEnum.targetUserName, username);
+ } catch (e) {
+     console.error('Cannot access LocalStorage', e);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->